core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -1624,6 +1626,24 @@ private RexNode simplifyOrs(List<RexNode> terms, RexUnknownAs unknownAs) {
             continue;
           }
         }
+        break;
+      case NOT_EQUALS:
+        Comparison notEqualsComparison =
+            Comparison.of(term, node -> RexUtil.isReferenceOrAccess(node, false));
```

Comment:
If it's not limited by `RexUtil.isReferenceOrAccess `, is it possible we support below case ?
```
(x+y <> a OR x+y <> b) => TRUE

Reply:
Theoretically yes, but in practice it does not even work right now in cases like:
`x+y = a AND x+y = b => could be simplified as FALSE, but it is not`
I think it would be too risky to try to simplify other things than ReferenceOrAccess, but I will let other experts in this module to confirm this. 
So, I'd rather keep this change simple and just enable a simplification for the scenario described above.

Reply:
I got it;
THX @rubenada  for reply :)